qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -2237,6 +2237,144 @@ def test_config_override(self):
                 except Exception as ex:
                     self.assertTrue(False, "unexpected error {}".format(ex))
 
+    def test_file_override(self):
+        import base64
+
+        model_shape = (1, 16)
+        override_base = "override_model"
+
+        for base in (('onnx', 'onnxruntime'),):
+            model_name = tu.get_model_name(base[0], np.float32, np.float32,
+                                           np.float32)
+            override_model_name = tu.get_model_name(override_base, np.float32,
+                                                    np.float32, np.float32)
+
+            # Prepare override file
+            with open("models/{}/3/model.{}".format(model_name, base[0]),
+                      'rb') as f:
+                encoded_file = base64.b64encode(f.read())
+
+            for triton_client in (httpclient.InferenceServerClient(
+                    "localhost:8000", verbose=True),
+                                  grpcclient.InferenceServerClient(
+                                      "localhost:8001", verbose=True)):
+                try:
+                    self.assertTrue(triton_client.is_server_live())
+                    self.assertFalse(
+                        triton_client.is_model_ready(model_name, "1"))
+                    self.assertFalse(
+                        triton_client.is_model_ready(model_name, "2"))
+                    self.assertTrue(
+                        triton_client.is_model_ready(model_name, "3"))
+                except Exception as ex:
+                    self.assertTrue(False, "unexpected error {}".format(ex))
+
+                # Request to load the model with override file, should fail
+                # without providing override config. The config requirement
+                # serves as an reminder that the existing model directory will
+                # not be used.
+                try:
+                    triton_client.load_model(
+                        model_name,
+                        encoded_files={"file:1/model.onnx": encoded_file})
+                    self.assertTrue(
+                        False, "expected error on missing override config")
+                except InferenceServerException as ex:
+                    # [FIXME] Improve error reporting to mention missing config
+                    self.assertIn(
+                        "failed to load '{}', failed to poll from model repository"
+                        .format(model_name), ex.message())
+
+                # Sanity check on previous loaded version is still available
```

Comment:
These comments are throwing me off a bit. So only one of the versions (3) should be available, right? After this, it looks like you load the version 1 and the original version is also available then... but 3 is not?

Reply:
The load above it fails (file override requires config to be provided as well), so the model availability hasn't changed and version 3 should still be ready.